cmd/lotus-gateway/api.go

```diff
@@ -172,6 +177,14 @@ func (a *GatewayAPI) StateWaitMsg(ctx context.Context, msg cid.Cid, confidence u
 	return a.api.StateWaitMsgLimited(ctx, msg, confidence, stateWaitLookbackLimit)
 }
 
+func (a *GatewayAPI) WalletVerify(ctx context.Context, k address.Address, msg []byte, sig *crypto.Signature) (bool, error) {
+	return sigs.Verify(sig, k, msg) == nil, nil
+}
+
+func (a *GatewayAPI) StateReadState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*api.ActorState, error) {
```

Comment:
In lite-mode, we use dependency injection to replace some of the methods in StateAPI, ChainAPI etc with methods in GatewayAPI. The way this works for StateAPI is:
- The method is added to the StateModuleAPI
- The method is moved from StateAPI to StateModule (Note that StateModule extends StateAPI)
- GatewayAPI implements a method of the same name
- In lite-mode Dependency Injection replaces StateModule with GatewayAPI

So if you add `StateReadState` to GatewayAPI you'll probably also want to add it to [node/impl/full/state.go StateModuleAPI](https://github.com/filecoin-project/lotus/blob/feat/gateway-wallet-verify/node/impl/full/state.go#L42) and then change [StateAPI.StateReadAPI to be StateModule.StateReadAPI](https://github.com/filecoin-project/lotus/blob/feat/gateway-wallet-verify/node/impl/full/state.go#L425):
```
- func (a *StateAPI) StateReadState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*api.ActorState, error) {
+ func (m *StateModule) StateReadState(ctx context.Context, actor address.Address, tsk types.TipSetKey) (*api.ActorState, error) {
```

[See StateAccountKey](https://github.com/filecoin-project/lotus/blob/feat/gateway-wallet-verify/node/impl/full/state.go#L416) for an example.

Reply:
@dirkmc To make sure we're on the same page, what youre proposing is adding support for lotus-lite to call directly to the gateway to use StateReadState. I'm not currently focused on lotus-lite, what does it do right now for calls to StateReadState?

Reply:
Right now it will just call its local StateReadState, which probably won't return anything very interesting because lotus-lite doesn't sync the chain.
I don't believe the change in this PR will have any effect on lotus-lite